Documentation improvement

alixdamman opened the pull request:
Still need to find a solution for larrayenv (conda-forge)

gdementen commented on larray/example.py:
```diff
@@ -0,0 +1,43 @@
+
+EXAMPLE_FILES_DIR = os.path.dirname(__file__) + '/tests/data/'
+
+def load_example_arrays():
```
We should have an argument to this function because we will probably include several example datasets in the future. The goal was to have something akin to http://www.stata.com/help.cgi?sysuse. The question is whether pop+hh+qx is a single "Session" dataset, or if those are three individual LArray dataset which happen to be somewhat linked. I would lean towards the former, but then we need a name for the global thing :)

gdementen replied:
also, I never liked that hh array. We cannot combine it with the two others in anyway and it is not super interesting by itself, so I do not think it justifies its place in the tutorial.

gdementen commented on doc/source/changes/version_0_22.rst.inc:
```diff
@@ -90,4 +90,4 @@ Fixes
 * fixed group aggregates on integer arrays for median, percentile, var and std (closes :issue:`193`).
 
-* fixed group sum over boolean arrays (closes :issue:`194`).
+* fixed set_labels when inplace=True
```
bad merge?

alixdamman replied:
yes

gdementen commented on larray/__init__.py:
```diff
@@ -6,6 +6,8 @@
 from larray.excel import open_excel
 from larray.ipfp import ipfp
 
+from larray.example import load_example_data
```
this has nothing to do with the set_labels fix

gdementen commented on larray/example.py:
```diff
@@ -0,0 +1,54 @@
+import os
```
this has nothing to do with the set_labels fix

gdementen commented on larray/example.py:
```diff
@@ -0,0 +1,54 @@
+
+    res = la.Session(AVAILABLE_EXAMPLE_DATA[example_data])
+
+    return res
```
inline res?

gdementen commented on larray/example.py:
```diff
@@ -0,0 +1,54 @@
+    'demography' : EXAMPLE_FILES_DIR + 'data.h5'
+}
+
+def load_example_data(example_data=None):
```
I am unsure having a default value is a good idea? Wouldn't "name" be a better argument name?

gdementen commented on larray/example.py:
```diff
@@ -0,0 +1,54 @@
+    example_data : str
+        Example data to load. Available example datasets are:
+
+        - demography
```
Wouldn't it be better to populate this list automatically, so that we do not have to keep it in sync?

gdementen commented on larray/example.py:
```diff
@@ -0,0 +1,54 @@
+        raise TypeError("Expected string for argument example_data")
+    if example_data not in AVAILABLE_EXAMPLE_DATA.keys():
+        raise ValueError("example_data must be chosen "
+                         "from list {}".format(list(AVAILABLE_EXAMPLE_DATA.keys())))
```
That's a good idea to give the list of possible values!

alixdamman commented:
I cleaned my history and removed the commit where I replace gdmenten by conda-forge in README.rst from this PR. I don't know how much time it will take to push larrayenv on conda-forge.

gdementen commented:
LGTM, thanks!
